### PR TITLE
Add wheels build for Python SDK and publish it to PyPI

### PR DESCRIPTION
**Pull request by t0yv0**
This is take 2 of the Python wheels rollout.

Fixes https://github.com/pulumi/pulumi-azure-native/issues/2476

### CODE REVIEW

File: provider/pkg/gen/__snapshots__/gen_vnet_test.snap [hunk 1]
```diff
@@ -20,7 +20,7 @@
...
...0x79, 0x20, 0x41, 0x7a, 0x75, 0x72, 0x65, 0x20, 0x72, 0x65, 0x73, 0x6f, 0x75, 0x72, 0x63, 0x65, 0x20, 0x61, 0x6e, 0x64, 0x20, 0x61, 0x6e, 0x79, 0x20, 0x73, 0x75, 0x70, 0x70, 0x6f, 0x72, 0x74, 0x65, 0x64, 0x20, 0x72, 0x65, 0x73, 0x6f, 0x75, 0x72, 0x63, 0x65, 0x5c, 0x6e, 0x76, 0x65, 0x72, 0x73, 0x69, 0x6f, 0x6e, 0x20, 0x75, 0x73, 0x69, 0x6e, 0x67, 0x20, 0x69, 0x6e, 0x66, 0x72, 0x61, 0x73, 0x74, 0x72, 0x75, 0x63, 0x74, 0x75, 0x72, 0x65, 0x20, 0x61, 0x73, 0x20, 0x63, 0x6f, 0x64, 0x65, 0x2c, 0x20, 0x77, 0x68, 0x69, 0x63, 0x68, 0x20, 0x50, 0x75, 0x6c, 0x75, 0x6d, 0x69, 0x20, 0x74, 0x68, 0x65, 0x6e, 0x20, 0x75, 0x73, 0x65, 0x73, 0x20, 0x74, 0x6f, 0x20, 0x64, 0x72, 0x69, 0x76, 0x65, 0x20, 0x74, 0x68, 0x65, 0x20, 0x41, 0x52, 0x4d, 0x20, 0x41, 0x50, 0x49, 0x2e, 0x22, 0x2c, 0x22, 0x72, 0x65, 0x71, 0x75, 0x69, 0x72, 0x65, 0x73, 0x22, 0x3a, 0x7b, 0x22, 0x70, 0x75, 0x6c, 0x75, 0x6d, 0x69, 0x22, 0x3a, 0x22, 0x5c, 0x75, 0x30, 0x30, 0x33, 0x65, 0x3d, 0x33, 0x2e, 0x33, 0x35, 0x2e, 0x30, 0x2c, 0x5c, 0x75, 0x30, 0x30, 0x33, 0x63, 0x34, 0x2e, 0x30, 0x2e, 0x30, 0x22, 0x7d, 0x2c, 0x22, 0x75, 0x73, 0x65, 0x73, 0x49, 0x4f, 0x43, 0x6c, 0x61, 0x73, 0x73, 0x65, 0x73, 0x22, 0x3a, 0x74, 0x72, 0x75, 0x65, 0x7d},
```
Comment (thomas11) [inline provider/pkg/gen/__snapshots__/gen_vnet_test.snap:23]: shouldn't the 13th byte be 0x3b?

File: sdk/python/pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+[project]
+  name = "pulumi_azure_native"
+  description = "A native Pulumi package for creating and managing Azure resources."
+  dependencies = ["parver>=0.2.1", "pulumi>=3.35.0,<4.0.0", "semver>=2.8.1"]
+  keywords = ["pulumi", "azure", "azure-native", "category/cloud", "kind/native"]
+  readme = "README.md"
+  requires-python = ">=3.7"
```
Comment (thomas11) [inline sdk/python/pyproject.toml:7]: This repo's CI uses Python 3.8 - doesn't mean 3.7 won't work but we're not testing it.

**Review by thomas11 — APPROVED** [review]

File: sdk/python/pyproject.toml (hunk 2 above)
Comment (t0yv0) [reply sdk/python/pyproject.toml:7]: This is hopefully out of scope for this change.. the pyproject.toml is just a projection of what's in the schema, just like setup.py was.

File: provider/pkg/gen/__snapshots__/gen_vnet_test.snap (hunk 1 above)
Comment (t0yv0) [reply provider/pkg/gen/__snapshots__/gen_vnet_test.snap:23]: That's the downside of binary snapshots. What's changing is that "pyproject.toml" enablement bit is now part of language-specific schema bits.

**Pull request merged**